aleth-bootnode/main.cpp

```diff
@@ -0,0 +1,129 @@
+using namespace std;
+
+int main(int argc, char** argv)
+{
+    setDefaultOrCLocale();
+
+    string const ProgramName = "aleth-bootnode";
+    string const NetworkConfigFileName = ProgramName + "-network.rlp";
+
+    /// Networking params.
+    string listenIP;
+    unsigned short listenPort = c_defaultIPPort;
+    string publicIP;
+    bool upnp = true;
+
+    po::options_description generalOptions("GENERAL OPTIONS", lineWidth());
+    auto addGeneralOption = generalOptions.add_options();
+    addGeneralOption("help,h", "Show this help message and exit\n");
+
+    LoggingOptions loggingOptions;
+    po::options_description loggingProgramOptions(
+        createLoggingProgramOptions(lineWidth(), loggingOptions));
+
+    po::options_description clientNetworking("NETWORKING", lineWidth());
+    auto addNetworkingOption = clientNetworking.add_options();
+#if ETH_MINIUPNPC
+    addNetworkingOption(
+        "upnp", po::value<string>()->value_name("<on/off>"), "Use UPnP for NAT (default: on)");
+#endif
+    addNetworkingOption("public-ip", po::value<string>()->value_name("<ip>"),
+        "Force advertised public IP to the given IP (default: auto)");
+    addNetworkingOption("listen-ip", po::value<string>()->value_name("<ip>(:<port>)"),
+        "Listen on the given IP for incoming connections (default: 0.0.0.0)");
+    addNetworkingOption("listen", po::value<unsigned short>()->value_name("<port>"),
+        "Listen on the given port for incoming connections (default: 30303)\n");
+
+    po::options_description allowedOptions("Allowed options");
+    allowedOptions.add(generalOptions).add(loggingProgramOptions).add(clientNetworking);
+
+    po::variables_map vm;
+    try
+    {
+        po::parsed_options parsed = po::parse_command_line(argc, argv, allowedOptions);
+        po::store(parsed, vm);
+        po::notify(vm);
+    }
+    catch (po::error const& e)
+    {
+        cout << e.what() << "\n";
+        return -1;
+    }
+
+    if (vm.count("help"))
+    {
+        cout << "NAME:\n"
+             << "   " << ProgramName << "\n"
+             << "USAGE:\n"
+             << "   " << ProgramName << " [options]\n\n";
+        cout << generalOptions << clientNetworking << loggingProgramOptions;
+        return 0;
+    }
+    if (vm.count("public-ip"))
+        publicIP = vm["public-ip"].as<string>();
+    if (vm.count("listen-ip"))
+        listenIP = vm["listen-ip"].as<string>();
+    if (vm.count("listen"))
+        listenPort = vm["listen"].as<unsigned short>();
+
+    setupLogging(loggingOptions);
+    if (loggingOptions.verbosity > 0)
+        cout << EthGrayBold << ProgramName << ", a C++ Ethereum bootnode implementation" EthReset
+             << "\n";
+
+    auto netPrefs = publicIP.empty() ? NetworkConfig(listenIP, listenPort, upnp) :
+                                       NetworkConfig(publicIP, listenIP, listenPort, upnp);
+    auto netData = contents(getDataDir() / fs::path(NetworkConfigFileName));
+
+    // TODO: Compose client version
```

Comment:
Not that much important for the bootnode I think (it's used in RLPx handshake, but we'll return no capabilities and it won't go any further)

Reply:
Got it, that makes things simpler (no need for buildinfo.h). Thanks for the tip!